Feature/notification class api improvements

meamuri opened the pull request:
Changes:
- jvm target version up to 8
- add notification builder and constructor with ttl
- write some test for notification class

meamuri committed:
add extra constructor with ttl and key primitives (strings and byte[]) to notification class

meamuri committed:
add builder to notificatino class

meamuri committed:
write tests for notification class

meamuri committed:
add overloads for publicKey and auth setters of builder

martijndwars commented:
Thanks for the contribution, great work! Do you think you can keep the source/target Java 7-compatible? I'd hate to disappoint Java 7 users (which, surprisingly, is still a thing in many companies).

meamuri committed:
revert compatibility with java 7 (keep target 7)

meamuri commented:
Thank you for your feedback! Yep, I updated version to 8 only to make it able for Duration api usage.
It isn't necessary and I reverted this changes. Take a look at current version please 😄

martijndwars commented:
Looks great, thanks! I'll release a new version to Maven Central in the next few days.
